tensorflow/core/ops/nn_ops.cc

```diff
@@ -1102,45 +1112,93 @@ REGISTER_OP("MaxPool3D")
     .Attr("ksize: list(int) >= 5")
     .Attr("strides: list(int) >= 5")
     .Attr(GetPaddingAttrString())
-    .Attr("T: numbertype")
+    .Attr(GetConvnetDataFormatAttrString())
+    .Attr("T: numbertype = DT_FLOAT")
     .SetShapeFn(shape_inference::Pool3DShape)
     .Doc(R"doc(
 Performs 3D max pooling on the input.
 
-ksize: 1-D tensor of length 5. The size of the window for each dimension of
-  the input tensor. Must have `ksize[0] = ksize[4] = 1`.
-strides: 1-D tensor of length 5. The stride of the sliding window for each
-  dimension of `input`. Must have `strides[0] = strides[4] = 1`.
+ksize: The size of the window for each dimension of the input tensor.
+strides: The stride of the sliding window for each dimension of the
+  input tensor.
 padding: The type of padding algorithm to use.
+data_format: Specify the data format of the input and output data. With the
+    default format "NHWC", the data is stored in the order of:
+        [batch, in_planes, in_height, in_width, in_channels].
+    Alternatively, the format could be "NCHW", the data storage order of:
+        [batch, in_channels, in_planes, in_height, in_width].
 input: Shape `[batch, depth, rows, cols, channels]` tensor to pool over.
 output: The max pooled output tensor.
 )doc");
 
 REGISTER_OP("MaxPool3DGrad")
-    .Input("orig_input: float")
-    .Input("orig_output: float")
+    .Input("orig_input: TInput")
```

Comment:
It is related to compatibility tests. Previous registration uses `orig_input: float` and `grad: T`. If I try to generalize and set `orig_input: T`, the compatibility test will fail. It was previously discussed here - https://github.com/tensorflow/tensorflow/pull/6664#discussion_r96332231 and I followed @vrv suggestion.